ecs-init/docker/docker_test.go

```diff
@@ -351,6 +352,45 @@ func TestStartAgentEnvFile(t *testing.T) {
 		t.Error("Error should not be returned")
 	}
 }
+func TestStartAgentWithGPUConfig(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	envFile := "\nECS_ENABLE_GPU_SUPPORT=true\n"
+	containerID := "container id"
+	expectedAgentBinds += 1
+
+	defer func() {
+		expectedAgentBinds = expectedAgentBindsUnspecifiedPlatform
+	}()
+
+	mockFS := NewMockfileSystem(mockCtrl)
+	mockDocker := NewMockdockerclient(mockCtrl)
+
+	mockFS.EXPECT().ReadFile(config.InstanceConfigFile()).Return([]byte(envFile), nil).AnyTimes()
+	mockFS.EXPECT().ReadFile(config.AgentConfigFile()).Return(nil, errors.New("not found")).AnyTimes()
+	mockDocker.EXPECT().CreateContainer(gomock.Any()).Do(func(opts godocker.CreateContainerOptions) {
+		validateCommonCreateContainerOptions(opts, t)
```

Comment:
How did you verify that the GPU dir is actually binded here? Is `validateCommonCreateContainerOptions` already changed to verify the GPU bind mount?

Reply:
in all these tests, we just verify the number of bind mounts rather than actual value. I'll check if we can verify GPU dir mount.

Reply:
I think it's validated in `validateCommonCreateContainerOptions`

https://github.com/haikuoliu/amazon-ecs-init/blob/master/ecs-init/docker/docker_test.go#L249

Reply:
updated